### PR TITLE
Implementation of CPOP scheduling policy

### PR DESCRIPTION
**Pull request by Thomasdezeeuw**

### CODE REVIEW

**Comment by Thomasdezeeuw** [comment]
Rebased this on master.

File: opendc-model-odc/core/src/main/kotlin/com/atlarge/opendc/model/odc/platform/scheduler/stages/machine/Selection.kt [hunk 1]
```diff
@@ -120,6 +120,32 @@ class HeftMachineSelectionPolicy : MachineSelectionPolicy {
...
+    override suspend fun select(machines: List<Machine>, task: Task): Machine? =
+        context<StageScheduler.State, OdcModel>().run {
+            model.run {
+                // NOTE: higher is better.
+                fun communication(task: Task, machine: Machine): Double {
+                    val ethernet_speeds = machine.outgoingEdges.destinations<Double>("ethernet_speed")
+                    val ethernet_speed = ethernet_speeds.sum()
+                    return ethernet_speed.toDouble() / task.inputSize
+                }
+                fun available_compute(machine: Machine): Double {
```
Comment (fabianishere) [inline opendc-model-odc/core/src/main/kotlin/com/atlarge/opendc/model/odc/platform/scheduler/stages/machine/Selection.kt]: Could you update these files to use camel case?

File: opendc-model-odc/core/src/main/kotlin/com/atlarge/opendc/model/odc/platform/scheduler/stages/machine/Selection.kt [hunk 2]
```diff
@@ -120,6 +120,32 @@ class HeftMachineSelectionPolicy : MachineSelectionPolicy {
...
+ * Critical-Path-on-a-Processor (CPOP) scheduling as described by H. Topcuoglu et al. in
+ * "Task Scheduling Algorithms for Heterogeneous Processors".
+ */
+class CpopMachineSelectionPolicy : MachineSelectionPolicy {
+    override suspend fun select(machines: List<Machine>, task: Task): Machine? =
+        context<StageScheduler.State, OdcModel>().run {
+            model.run {
+                // NOTE: higher is better.
+                fun communication(task: Task, machine: Machine): Double {
+                    val ethernet_speeds = machine.outgoingEdges.destinations<Double>("ethernet_speed")
```
Comment (fabianishere) [inline opendc-model-odc/core/src/main/kotlin/com/atlarge/opendc/model/odc/platform/scheduler/stages/machine/Selection.kt]: Could you update this to use `machine.ethernetSpeed`?

**Comment by Thomasdezeeuw** [comment]
Fixed both issues.

**Review by fabianishere — APPROVED** [review]

**Comment by Thomasdezeeuw** [comment]
@fabianishere thanks again for all your help throughout our lab assignment.

**Comment by fabianishere** [comment]
@Thomasdezeeuw Good job on these contributions!

**Pull request merged**